### DISCUSSION TITLE
SSL Handshake caching

### DISCUSSION THREAD

**Discussion by libertyforward1** [mastodon/mastodon#21836] [Q&A]
I apologize but this is a struggle for me to explain, please bear with me. I subscribed to a couple of relays for my Mastodon instance, and about 90% of the time, the server is throwing back an HTTP 401 error, seemingly due to timeout because it's asking for the key to be presented with every single contact rather than saving it to cache. Is this expected behavior, and if so, is there any way for me to avoid triggering the timeout? If it isn't expected, how might I go about alleviating the problem?

**Comment by nightpool** [comment]
The relays are misconfigured. The key requests are triggered *BY* the HTTP signature validation failing, not the other way around.

**Reply by libertyforward1** [reply]
Very interesting. The odd thing is, every so often I see the relay contact me and my server throws back response code 202. Is it possible the relay is expecting my system to cache and therefore presenting nothing at all - thereby triggering the fail?

**Comment by libertyforward1** [comment]
Turns out after many hours of frustration and research by the relay operator (his kindness is incredible!) it all comes down to me stupidly using camelCase to set up my instance. The single capitalized letter broke everything for connecting to relays. *sigh*

I'm guessing if I change .env.production to have the domain all in lowercase I'll break everything else, right?
